src/compiler/checker.ts

```diff
@@ -38322,7 +38322,7 @@ namespace ts {
             // For a binding pattern, validate the initializer and exit
             if (isBindingPattern(node.name)) {
                 const needCheckInitializer = hasOnlyExpressionInitializer(node) && node.initializer && node.parent.parent.kind !== SyntaxKind.ForInStatement;
-                const needCheckWidenedType = !some(node.name.elements, not(isOmittedExpression));
+                const needCheckWidenedType = !!node.name.elements.length && !some(node.name.elements, not(isOmittedExpression));

```

Comment:
actually - this is not a good fix, i need to check if this is contained in a declaration signature, gonna push out a proper fix some time later

Reply:
Yeah, I think if this is written outside of a declaration, it's always an error, because you can't have `{}?` or `{ x }?` without it crashing; the question token wouldn't be there if there were a default as that'd be `{} = {}` and so on.

Reply:
Ye, i’m looking throughout the codebase to see how other places check if a thing is in a type declaration but surprisingly i didnt find many. Im gonna resort to a findAncestor check

Reply:
Ok, I think I got it right now